CDAP-13857 Document metadata migration

CuriousVini opened the pull request:
Quick build: https://builds.cask.co/browse/CDAP-DQB420-1

CuriousVini committed:
CDAP-13857 Document metadata migration

deb-saunders commented on cdap-docs/reference-manual/source/http-restful-api/metadata.rst:
```diff
@@ -34,6 +34,11 @@ metadata object associated with a single entity is limited to 10K bytes in size.
 There is one reserved word for property keys and values: *tags*, either as ``tags`` or
 ``TAGS``. Tags themselves have no reserved words.
 
+Please note that CDAP will perform online metadata migration when CDAP is migrated from 4.3.x to 5.0.0. When metadata
```
**Note**: The online metadata is migrated when CDAP migrates from 4.3.x to 5.0.0.  During metadata migration, writes to non-migrated entities fail and reads on metadata for non-migrated entities do not have all the metadata. Metadata migration takes approximately 2-3 minutes. You can check the metadata migration status by using the :ref:`upgrade status <http-restful-api-monitor-status-system-upgrade>` REST endpoint.

GLOBAL: changing to present tense & active voice

deb-saunders commented on cdap-docs/reference-manual/source/http-restful-api/metadata.rst:
```diff
@@ -34,6 +34,11 @@ metadata object associated with a single entity is limited to 10K bytes in size.
 
+Please note that CDAP will perform online metadata migration when CDAP is migrated from 4.3.x to 5.0.0. When metadata
+migration is in progess, any writes to non-migrated entities will fail and reads on metadata for non-migrated entities
+will not have all the metadata. Metadata migration is expected to take couple of minutes. Please use
```
couple? 2-3 ... 5 -10? Can we timebox?

CuriousVini replied:
We have not specified any number here because the time would depend on how much data needs migration. However, we do not expect it to be too long.

CuriousVini replied:
Actually I think 2-3 minutes is a good number.

CuriousVini commented:
Thanks @deb-saunders For the quick review, I have addressed the comments in a separate commit. Please take a look at it when you get a chance, thanks!
